### ISSUE TITLE
Will this SVL work on Apollo3 Blue Plus?

### ISSUE THREAD

**Issue by wzqvip** [sparkfun/Apollo3_Uploader_SVL#9]
Apollo3Blue (AMA3B2KK) had doubled RAM and flash, the else are all the same. And it's hard to buy 3b today, 3b+ is cheaper and easier to find purchase.

Will the SVL work on 3b+? or how can I adapt it to 3b+.

There should be no huge difference between these two mcu.
By the way, I want to make it work on Apollo 4 too, can I just compile it for apollo 4?

**Comment by JaredSapphire** [comment]
Here is an SVL fork that will work with the Apollo3 Blue Plus Microcontroller.  https://github.com/SapphireCircuits/Apollo3Plus_Uploader_SVL

**Comment by dboddie** [comment]
@JaredSapphire Does v6 write correctly to the top half of the flash in an Apollo3 Blue? I have tried the latest version in the SparkFun repo but, while it erases pages in the top half, it doesn't seem to write data to them.

**Comment by JaredSapphire** [comment]
Yes.  The V6 in the branch https://github.com/SapphireCircuits/Apollo3Plus_Uploader_SVL/tree/flash-upper-half is the one you want.  The V7 is for the Apollo3 Blue Plus.

**Comment by JaredSapphire** [comment]
I commented too soon. It looks like the codes should be merged.  I'll test it out.

**Comment by JaredSapphire** [comment]
@dboddie The source files looked synced, atleast from my diffs.  Are you using the built svl.bin file or re-making from source?

**Comment by dboddie** [comment]
I'm using the built `svl.bin` file in the `gcc` subdirectory. I reflashed the bootloader using the Arduino IDE. It looks like it's identical to the one in your fork, in the relevant branch.

When I send a payload to the board using the `svl.py` script, it reports that it has the new bootloader:

    Artemis SVL Bootloader
    Got SVL Bootloader Version: 6

**Comment by JaredSapphire** [comment]
Updated: 
(original link was the main, instead of flash-upper-half branch) 
can you retry it with the one just updated here:
https://github.com/SapphireCircuits/Apollo3Plus_Uploader_SVL/tree/flash-upper-half/bootloader/gcc/artemis_module/bin

**Comment by dboddie** [comment]
Initial tests indicate that it worked. Thanks!
Should I make a new issue for this repository to get the bootloader updated with the latest v6?

**Comment by JaredSapphire** [comment]
I'm glad that version worked.  I just re-made them (again) and tested it on a SparkFun RedBoard Artemis Nano.  It should be something that gets looked into.

tagging @gigapod

**Comment by gigapod** [comment]
HI @JaredSapphire  - I haven't looked at the changes made in v7 mentioned above to pull in for our boards. Would this be helpful in our Arduino distro even though we don't support the Blue Plus version of the iC?  

I'm planning to push out an Apollo 3 Arduino release shortly when I have time - I could give this a try and if it works, roll it in.

**Comment by JaredSapphire** [comment]
The V7 is specific to the Sapphire Circuits Artemis Plus Module design (w/ Apollo3 Blue Plus micro) and as it stands now would most likely break compatibility with the SparkFun Artemis line.  A merge might be possible with more complex makefiles that check which microcontroller or platform is in use and sets the paths, defines, and source files accordingly, but at the moment, the makefiles are not set up that way.  

The V6 and the "flash-upper-half" branch was made specifically made to help SparkFun Artemis users, and to give back to SparkFun user community.   Thank you [SparkFun] for making the Artemis module and open sourcing it in the first place.  The Artemis Plus version which is a derivative work of the SparkFun Artemis would probably not have existed otherwise.

**Comment by gigapod** [comment]
Thanks @JaredSapphire  for the information - I wasn't aware of the Artemis Plus module!  

And thanks for the v6 update!  I'll get our update out over the holidays.

-Kirk
